src/job/models/ingestion/seedingJobCreator.ts

```diff
@@ -98,14 +96,87 @@ export class SeedingJobCreator {
     });
   }
 
-  private createSeedOptions(mode: SeedMode, geometry: Footprint, cacheName: string): SeedTaskOptions {
+  private handleCleanMode(
+    job: IngestionUpdateFinalizeJob | IngestionSwapUpdateFinalizeJob,
+    cacheName: string,
+    catalogId: string
+  ): { type: string; parameters: SeedTaskParams }[] {
+    const activeSpan = trace.getActiveSpan();
+    const seedTaskType = this.tilesSeedingConfig.type;
+    const logger = this.logger.child({ mode: SeedMode.CLEAN });
+
+    const cleanGeometry = this.calculateGeometryByMode(SeedMode.CLEAN, job);
+    if (!cleanGeometry) {
+      activeSpan?.addEvent('calculateCleanGeometry.empty');
+      logger.warn({ msg: 'No geometry found for CLEAN mode' });
+      return [];
+    }
+
+    activeSpan?.addEvent('calculateCleanGeometry.success', { geometry: JSON.stringify(cleanGeometry) });
+
+    const seedTasks: { type: string; parameters: SeedTaskParams }[] = [];
+
+    if (job.type === this.config.get<string>('jobManagement.polling.jobs.swapUpdate.type')) {
+      const cleanOptions = this.createSeedOptions(SeedMode.CLEAN, cleanGeometry, cacheName);
+      activeSpan?.addEvent('createSeedOptions.success', { seedOptions: JSON.stringify(cleanOptions) });
+
+      const taskParams = this.createTaskParams(catalogId, cleanOptions);
+      activeSpan?.addEvent('createTaskParams.success', { taskParams: JSON.stringify(taskParams) });
+
+      seedTasks.push({ type: seedTaskType, parameters: taskParams });
+    }
+
+    if (job.type === this.config.get<string>('jobManagement.polling.jobs.update.type')) {
+      const maxUpdatedZoom = extractMaximalUpdatedResolution(job);
+      if (maxUpdatedZoom + 1 <= this.tilesSeedingConfig.maxZoom) {
+        const cleanOptions = this.createSeedOptions(SeedMode.CLEAN, cleanGeometry, cacheName, maxUpdatedZoom + 1);
+        activeSpan?.addEvent('createSeedOptions.success', { seedOptions: JSON.stringify(cleanOptions) });
+
+        const taskParams = this.createTaskParams(catalogId, cleanOptions);
+        activeSpan?.addEvent('createTaskParams.success', { taskParams: JSON.stringify(taskParams) });
+
+        seedTasks.push({ type: seedTaskType, parameters: taskParams });
+      }
+    }
+
+    return seedTasks;
+  }
+
+  private handleSeedMode(
+    job: IngestionUpdateFinalizeJob | IngestionSwapUpdateFinalizeJob,
+    cacheName: string,
+    catalogId: string
+  ): { type: string; parameters: SeedTaskParams }[] {
```

Comment:
why an array returned if only a single { type: string; parameters: SeedTaskParams } item is returned